app/src/main/java/agency/five/tmdb/MovieApiImpl.kt

```diff
@@ -1,170 +1,90 @@
-    )
-
-    private val a3 = Actor(
-        movie = "Iron Man 1",
-        name = "Jeff",
-        surname = "Bridges",
-        role = "Obadiah Stane / Iron Monger",
-        imageResId = R.drawable.jeff_bridges
-    )
+class MovieApiImpl(
+    private val client: HttpClient
+) : MovieApi {
 
+    private val apiKey: String = "0d756c167ac7d8a9afae24e0d3af8a9e"
 
     override suspend fun getPopularMovies(): List<List<MovieItemViewState>> {   // size: 4 (tabs)
-        return listOf(l1, l2, l3, l1)
+
+        val forRent = getPopularForRent().subList(0, 10)
+
+        val inTheaters = getPopularInTheaters().subList(0, 10)
+
+        val onTV = getPopularOnTV().subList(0, 10)
+
+        val streaming = getPopularStreaming().subList(0, 10)
+
+        return listOf(streaming, onTV, forRent, inTheaters)
+    }
+
+    override suspend fun getPopularForRent(): List<MovieItemViewState> {
+        return client.get<MoviesResponse>("https://api.themoviedb.org/3/movie/popular?api_key=$apiKey").movies
+    }
+
+    override suspend fun getPopularInTheaters(): List<MovieItemViewState> {
+        return client.get<MoviesResponse>("https://api.themoviedb.org/3/movie/now_playing?api_key=$apiKey").movies
+    }
+
+    override suspend fun getPopularStreaming(): List<MovieItemViewState> {
+        // now playing movies on page=2
+        return client.get<MoviesResponse>("https://api.themoviedb.org/3/movie/now_playing?api_key=$apiKey&page=2").movies
+    }
+
+    override suspend fun getPopularOnTV(): List<MovieItemViewState> {
+        // popular movies on page=2
+        return client.get<MoviesResponse>("https://api.themoviedb.org/3/movie/popular?api_key=$apiKey&page=2").movies
     }
 
+
     override suspend fun getTrendingMovies(): List<List<MovieItemViewState>> {  // size: 2 (tabs)
-        return listOf(l2, l3)
+        return listOf(getTrendingToday().subList(0, 10), getTrendingThisWeek().subList(0, 10))
+    }
+
+    override suspend fun getTrendingToday(): List<MovieItemViewState> {
+        return client.get<MoviesResponse>("https://api.themoviedb.org/3/trending/movie/day?api_key=$apiKey").movies
+    }
+
+    override suspend fun getTrendingThisWeek(): List<MovieItemViewState> {
+        return client.get<MoviesResponse>("https://api.themoviedb.org/3/trending/movie/week?api_key=$apiKey").movies
     }
 
     override suspend fun getFreeMovies(): List<List<MovieItemViewState>> {  // size: 2 (tabs)
-        return listOf(l1, l3)
+        return listOf(getUpcomingMovies().subList(0, 10), getTopRatedMovies().subList(0, 10))
+    }
+
+    override suspend fun getTopRatedMovies(): List<MovieItemViewState> {
+        return client.get<MoviesResponse>("https://api.themoviedb.org/3/movie/top_rated?api_key=$apiKey").movies
+    }
+
+    override suspend fun getUpcomingMovies(): List<MovieItemViewState> {
+        return client.get<MoviesResponse>("https://api.themoviedb.org/3/movie/upcoming?api_key=$apiKey").movies
```

Comment:
The service/api classes are responsible only for fetching data from the server, you shouldn't have any additional logic like list combining, cutting etc.

You should move that logic in repository layer. (Also check the comment from previous PR about Data and Domain layer.)